files/setup_http_service.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# These variables must be set before client installation:
+#
+# hostname: The hostname of this IPA client (e.g. client.example.com).
+
+# The file installed by cloud-init that contains the value for the
+# above variables.
+freeipa_vars_file=/var/lib/cloud/instance/freeipa-vars.sh
+
+# Load above variable from a file installed by cloud-init:
+if [[ -f "$freeipa_vars_file" ]]
+then
+    # Disable this warning since the file is only available at runtime
+    # on the server.
+    #
+    # shellcheck disable=SC1090
+    source "$freeipa_vars_file"
+else
+    echo "FreeIPA variables file does not exist: $freeipa_vars_file"
+    echo "It should have been created by cloud-init at boot."
+    exit 254
+fi
+
+# Get the default Ethernet interface
+function get_interface {
+    ip route | grep default | sed "s/^.* dev \([^ ]*\).*$/\1/"
+}
+
+# Get the IP address corresponding to an interface
+function get_ip {
+    ip --family inet address show dev "$1" | \
+        grep --perl-regexp --only-matching 'inet \K[\d.]+'
+}
+
+# Create the HTTP/$hostname service if it does not already exist.
```

Comment:
I wasn't sure if this was intended to be the description for the entire script or just for this block, but if it was meant to be for the entire script, I'd prefer to see it near the top of the file.  If it wasn't, I'd still like to see a brief script description near the top of the file.

Reply:
See commit 019e6b7.